tests/aws-cpp-sdk-s3-unit-tests/S3UnitTests.cpp

```diff
@@ -178,3 +179,21 @@ TEST_F(S3UnitTest, S3UriPathPreservationOn) {
   const auto seenRequest = _mockHttpClient->GetMostRecentHttpRequest();
   EXPECT_EQ("https://velvetunderground.s3.us-east-1.amazonaws.com/////stephanie////says////////////that////////she//wants///////to/know.txt", seenRequest.GetUri().GetURIString());
 }
+
+TEST_F(S3UnitTest, S3EmbeddedErrorTest) {
+  const auto request = CopyObjectRequest()
+    .WithBucket("testBucket")
+    .WithKey("testKey")
+    .WithCopySource("testSource");
+
+  auto mockRequest = Aws::MakeShared<Standard::StandardHttpRequest>(ALLOCATION_TAG, "mockuri", HttpMethod::HTTP_GET);
+  mockRequest->SetResponseStreamFactory([]() -> IOStream* {
+    return Aws::New<StringStream>(ALLOCATION_TAG, "<Error><Code>SlowDown</Code><Message>Please reduce your request rate.</Message></Error>", std::ios_base::in | std::ios_base::binary);
```

Comment:
looks ok to me. One thing I noted is that the xml header is missing 
<?xml version=\"1.0\" encoding=\"UTF-8\"?>

Not sure if this is an issue. Otherwise looks fine.

Reply:
updated, good eye